test/controllers/user.controller.test.js

```diff
@@ -1240,4 +1247,119 @@ describe('User Controller', () => {
       });
     });
   });
+
+  describe('Account Overview', () => {
+    let token;
+    const userId = mongoose.Types.ObjectId();
+    const vendorId = mongoose.Types.ObjectId();
+    const propertyId = mongoose.Types.ObjectId();
+    const offerId = mongoose.Types.ObjectId();
+    const transactionId = mongoose.Types.ObjectId();
+    const user = UserFactory.build({ _id: userId });
+    const vendor = UserFactory.build({ _id: vendorId });
+    const property = PropertyFactory.build({
+      _id: propertyId,
+      addedBy: vendorId,
+      updatedBy: vendorId,
+      price: 20000000,
+    });
+    const referralId = mongoose.Types.ObjectId();
+    const referral = ReferralFactory.build({
+      _id: referralId,
+      referrerId: userId,
+      reward: { amount: 50000 },
+    });
+
+    const enquiryId = mongoose.Types.ObjectId();
+    const enquiry = EnquiryFactory.build({ _id: enquiryId, userId, propertyId });
+    const offer = OfferFactory.build({
+      _id: offerId,
+      enquiryId,
+      vendorId,
+      userId,
+      totalAmountPayable: 19000000,
+    });
+    const transaction = TransactionFactory.build({
+      _id: transactionId,
+      propertyId,
+      userId,
+      adminId: vendorId,
+      amount: 250000,
+    });
+
+    beforeEach(async () => {
+      token = await addUser(user);
+      await addUser(vendor);
+    });
+
+    context('with valid token', () => {
+      it('returns contribution reward of zero', (done) => {
+        request()
+          .get('/api/v1/user/account-overview')
+          .set('authorization', token)
+          .end((err, res) => {
+            expect(res).to.have.status(200);
+            expect(res.body.success).to.be.eql(true);
+            expect(res.body.accountOverview.contributionReward).to.be.eql(0);
+            expect(res.body.accountOverview.totalAmountPaid).to.be.eql(0);
+            expect(res.body.accountOverview.referralRewards).to.be.eql(0);
+            done();
+          });
+      });
+    });
+
+    describe('when offer has been accepted', () => {
+      beforeEach(async () => {
+        await addReferral(referral);
+        await addProperty(property);
+        await addEnquiry(enquiry);
+        await createOffer(offer);
+        await addTransaction(transaction);
+        await acceptOffer({ userId, offerId, signature: 'https://ballers.ng/signature.png' });
+      });
+
+      context('with valid token', () => {
+        it('returns a valid contribution reward', (done) => {
+          request()
+            .get('/api/v1/user/account-overview')
+            .set('authorization', token)
+            .end((err, res) => {
+              expect(res).to.have.status(200);
+              expect(res.body.success).to.be.eql(true);
+              expect(res.body.accountOverview.contributionReward).to.be.eql(1000000);
```

Comment:
Magic numbers are very difficult to understand when reading. This can be improved to something more meaningful like:

```suggestion
              expect(res.body.accountOverview.contributionReward).to.be.eql(property.price - offer.totalAmountPayable);
```